server/routes/appointments/appointmentSummary.ts

```diff
@@ -3,43 +3,68 @@ import AppointmentDecorator from '../../decorators/appointmentDecorator'
 import PresenterUtils from '../../utils/presenterUtils'
 import AuthUserDetails from '../../models/hmppsAuth/authUserDetails'
 import { InitialAssessmentAppointment, ActionPlanAppointment } from '../../models/appointment'
+import User from '../../models/hmppsAuth/user'
+import { AppointmentDeliveryType } from '../../models/appointmentDeliveryType'
+import Address from '../../models/address'
 
 export default class AppointmentSummary {
   constructor(
     private readonly appointment: InitialAssessmentAppointment | ActionPlanAppointment,
-    private readonly assignedCaseworker: AuthUserDetails | null = null
+    private readonly assignedCaseworker: AuthUserDetails | User | null = null
```

Comment:
I'm trying to figure out if there's a way we could only pass one of `User` or `AuthUserDetails` in here as I feel like checking the type of something e.g. based on it having a `firstName` field feels like a bit of a code smell - it might take a bit of refactoring the rest of the codebase (or not be possible, in which case, apologies), but what do you think?

We could potentially, as a first step, convert an `AuthUserDetails` into a `User` (or vice versa) and pass it in?

Reply:
I don't think it should be converted into `User` as that can be confusing for the caller.
What do you think about passing in the name into the constructor rather than an `AuthUserDetails` or `User` ?

Reply:
Yes, you're right....

I've created a common interface:

```
interface Caseworker {
  username?: string
  firstName?: string
  lastName?: string
}
```